pyahocorasick fails to build with GCC 15.0.1

``` console
$ podman run --rm -ti fedora:rawhide bash
$ dnf install -y python3-devel gcc
$ python3 -m venv venv
$ . ./venv/bin/activagte
$ pip install -v pyahocorasick
[...]
Building wheels for collected packages: pyahocorasick
  Running command Building wheel for pyahocorasick (pyproject.toml)
  running bdist_wheel
  running build
  running build_ext
  building 'ahocorasick' extension
  creating build/temp.linux-x86_64-cpython-313/src
  gcc -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -O3 -fPIC -DAHOCORASICK_UNICODE= -I/root/venv/include -I/usr/include/python3.13 -c src/pyahocorasick.c -o build/temp.linux-x86_64-cpython-313/src/pyahocorasick.o
  In file included from src/pyahocorasick.c:13:
  src/common.h:95:17: error: ‘bool’ cannot be defined via ‘typedef’
     95 | typedef char    bool;
        |                 ^~~~
  src/common.h:95:17: note: ‘bool’ is a keyword with ‘-std=c23’ onwards
  src/common.h:95:1: warning: useless type name in empty declaration
     95 | typedef char    bool;
        | ^~~~~~~
  error: command '/usr/bin/gcc' failed with exit code 1
  error: subprocess-exited-with-error
  
  × Building wheel for pyahocorasick (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /root/venv/bin/python3 /root/venv/lib64/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_wheel /tmp/tmpf0agygg6
  cwd: /tmp/pip-install-929tzrwi/pyahocorasick_9716e16ac3d4457b931218cd6384b10d
  Building wheel for pyahocorasick (pyproject.toml) ... error
  ERROR: Failed building wheel for pyahocorasick
Failed to build pyahocorasick
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (pyahocorasick)``
$ gcc --version
gcc (GCC) 15.0.1 20250114 (Red Hat 15.0.1-0)
```

--- a/src/common.h
+++ b/src/common.h
@@ -16,6 +16,7 @@
 #include <structmember.h>   // PyMemberDef
 
 #include <iso646.h>
+#include <stdbool.h>
 
 #define DEBUG
 
@@ -91,9 +92,5 @@
 #else
 #   define F(name) name
 #endif
-
-typedef char    bool;
-#define true 1
-#define false 0
 
 #endif

